src/media/js/compat_filter.js

```diff
@@ -69,7 +71,10 @@ define('compat_filter',
     // On desktop, the default is no filtering.
     if (actualPlatform == 'desktop') {
         actualPlatform = '';
-        filterDevice = '';
+
+        if (!storage.getItem(filterDeviceLSKey)) {
```

Comment:
This `storage.getItem()` call is done twice. Can you refactor the code in a way that would prevent that ?